catroid/src/main/java/org/catrobat/catroid/stage/StageResourceHolder.java

```diff
@@ -367,7 +367,12 @@ public void onClick(DialogInterface dialog, int id) {
 
 		if (requiredResourcesSet.contains(Brick.COLLISION)) {
 			GatherCollisionInformationTask task = new GatherCollisionInformationTask(this);
-			task.execute();
+
+			try {
+				task.execute().get();
```

Comment:
if there is an exception, at which point is it expected to happen? is onPostExecute of the thread called in that case? if not the stage will not start since listener.onFinished(); is never called and StageResourceHolder will never finish.

Reply:
i'm sry i fucked that one up that shouldn't be there anymore since the stage doesn't start anyway without having all resources